### PR TITLE
Turn on dependabot to roll bundler dependencies

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Turn on [GitHub dependabot](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-dependencies-updated-automatically) to keep the `dev/ci` `Gemfile.lock` files up to date.  Run it weekly.

## Related Issues

See https://github.com/flutter/flutter/pull/71531 where this was done manually.
It's now opt-in, but for a brief window we were getting dependabot PRs, see https://github.com/flutter/flutter/pull/62379

## Tests

Might not know if this works until next Monday.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "bundler"
+    directory: "/dev/ci"
+    schedule:
+      interval: "weekly"
+    reviewers:
+      - "jmagman"
```
Comment (jmagman) [inline .github/dependabot.yml:11]: @godofredoc Should someone from infra be added as a backup?

Comment (godofredoc) [reply .github/dependabot.yml:11]: sounds good, please add @keyonghan and @yusufm

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      - "keyonghan"
+      - "yusufm"
+    labels:
+      - "team"
+      - "team: infra"
+      - "waiting for tree to go green"
```
Comment (jmagman) [inline .github/dependabot.yml:17]: Will add [dependabot](https://github.com/flutter/cocoon/blob/70ba45e6998175aa79c2ad912fb9e6e8c1ba3299/app_dart/lib/src/datastore/cocoon_config.dart#L246-L247) to the autorollers list so we can mostly ignore it if the tests pass.

Comment (christopherfujino) [inline .github/dependabot.yml:17]: ;)

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**